homeassistant/components/vallox/services.yaml

```diff
@@ -27,3 +27,28 @@ set_profile_fan_speed_boost:
           min: 0
           max: 100
           unit_of_measurement: "%"
+
+set_profile:
+  fields:
+    profile:
+      required: true
+      selector:
+        select:
+          options:
+            - label: "Home"
+              value: "1"
+            - label: "Away"
+              value: "2"
+            - label: "Boost"
+              value: "3"
+            - label: "Fireplace"
+              value: "4"
+            - label: "Extra"
+              value: "5"
```

Comment:
So I am wondering, would this be able to be split up to a select entity and a number entity? I think the select entity would work, but I won't be sure if the logic around it could be reflected correctly, wdyt?

Reply:
`set_profile` method in the library supports setting duration. It works only for `fireplace` and `extra` profiles.

Reply:
So I think a single set_profile service would be fine accepting profile string name and optional duration. Creating int mappings feels wrong.

Reply:
I mean its not about int mapping, its more about moving to a service or to entities, as entities are more user friendly and accesible, the only question is, is this a good idea for these

Reply:
not that familiar with python, is there a simple way of having the parameter be coerced into the matching int enum of `Profile` from the lib?

or how would the mapping be done? e.g. from `"Boost"` to `Profile(3)`

Reply:
Oh but that's not the thing I am worried about, you can indeed do `Profile(3)`. I am more wondering if it fits and if it can be done directly.

Reply:
two parallel conversations happening I think :)

I'm not sure about either, my end goal is just to have a service to call from nodered to enable the programmable "extra" profile when my range hood turns on.

But the `vallox_websocket_api` makes interactions a bit easier by hiding some of the lower level stuff. Separating this into two entities might mean splitting some of the existing stuff to use the lower level modbus registers.

Like all the modes that allow timing, have separate registers for timing, and also a register for enabling or disabling the timers. And when setting a profile, it clears the existing timers for other profiles. Out of the box there is a priority order the device runs down.

I would keep everything close to the api which is less likely to change.